pkg/generated/restapi/configure_rekor_server.go

```diff
@@ -84,49 +84,75 @@ func configureAPI(api *operations.RekorServerAPI) http.Handler {
+	}
 
-	if viper.GetBool("enable_retrieve_api") {
-		api.IndexSearchIndexHandler = index.SearchIndexHandlerFunc(pkgapi.SearchIndexHandler)
-	} else {
-		api.IndexSearchIndexHandler = index.SearchIndexHandlerFunc(pkgapi.SearchIndexNotImplementedHandler)
+	for _, enabledAPI := range enabledAPIEndpoints {
+		log.Logger.Infof("Enabling API endpoint: %s", enabledAPI)
+		switch enabledAPI {
+		case "searchIndex":
+			api.IndexSearchIndexHandler = index.SearchIndexHandlerFunc(pkgapi.SearchIndexHandler)
+		case "getLogInfo":
+			api.TlogGetLogInfoHandler = tlog.GetLogInfoHandlerFunc(pkgapi.GetLogInfoHandler)
+		case "getPublicKey":
+			api.PubkeyGetPublicKeyHandler = pubkey.GetPublicKeyHandlerFunc(pkgapi.GetPublicKeyHandler)
+		case "getLogProof":
+			api.TlogGetLogProofHandler = tlog.GetLogProofHandlerFunc(pkgapi.GetLogProofHandler)
+		case "createLogEntry":
+			api.EntriesCreateLogEntryHandler = entries.CreateLogEntryHandlerFunc(pkgapi.CreateLogEntryHandler)
+		case "getLogEntryByIndex":
+			api.EntriesGetLogEntryByIndexHandler = entries.GetLogEntryByIndexHandlerFunc(pkgapi.GetLogEntryByIndexHandler)
+		case "getLogEntryByUUID":
+			api.EntriesGetLogEntryByUUIDHandler = entries.GetLogEntryByUUIDHandlerFunc(pkgapi.GetLogEntryByUUIDHandler)
+		case "searchLogQuery":
+			api.EntriesSearchLogQueryHandler = entries.SearchLogQueryHandlerFunc(pkgapi.SearchLogQueryHandler)
+		default:
+			log.Logger.Panicf("Unknown API endpoint requested: %s", enabledAPI)
+		}
 	}
 
+	// all handlers need to be set before a call to api.AddMiddlewareFor
+	for _, enabledAPI := range enabledAPIEndpoints {
+		switch enabledAPI {
+		case "searchIndex":
+			recordMetricsForAPI(api, "POST", "/api/v1/index/retrieve") // add metrics
+		case "getLogInfo":
+			api.AddMiddlewareFor("GET", "/api/v1/log", middleware.NoCache) // not cacheable
+			recordMetricsForAPI(api, "GET", "/api/v1/log")                 // add metrics
+		case "getPublicKey":
+			api.AddMiddlewareFor("GET", "/api/v1/log/publicKey", middleware.NoCache) // not cacheable
+			recordMetricsForAPI(api, "GET", "/api/v1/log/publicKey")                 // add metrics
+		case "getLogProof":
+			api.AddMiddlewareFor("GET", "/api/v1/log/proof", middleware.NoCache) // not cacheable
+			recordMetricsForAPI(api, "GET", "/api/v1/log/proof")                 // add metrics
+		case "createLogEntry":
+			recordMetricsForAPI(api, "POST", "/api/v1/log/entries") // add metrics
+		case "getLogEntryByIndex":
+			api.AddMiddlewareFor("GET", "/api/v1/log/entries", middleware.NoCache) // not cacheable
+			recordMetricsForAPI(api, "GET", "/api/v1/log/entries")                 // add metrics
+		case "getLogEntryByUUID":
+			api.AddMiddlewareFor("GET", "/api/v1/log/entries/{entryUUID}", middleware.NoCache) // not cacheable
+			recordMetricsForAPI(api, "GET", "/api/v1/log/entries/{entryUUID}")                 // add metrics
+		case "searchLogQuery":
+			recordMetricsForAPI(api, "POST", "/api/v1/log/entries/retrieve") // add metrics
```

Comment:
Thanks for correcting this from GET to POST, we're missing metrics because of this.